script.js

```diff
@@ -0,0 +1,134 @@
+
+let paginatedBtnGroup= document.querySelector(".paginated-btn-grp");
+// let currentPageNumber=1;
+let noOfPosts=6;
+let totalPage =0;
+let dataList=[];
+let currentPage=1;
+
+async function getAllData(){
+    
+    const data=await fetch('https://dummyjson.com/posts');
+    const json=await data.json();
+    //  dataList= json.posts;
+      dataList=json.posts;
+     console.log(dataList);
+
+     totalPage=Math.ceil(dataList.length / noOfPosts);
+     
+     createPaginatedButton(totalPage);
+     displayItems(dataList,currentPage); 
+
+    
+}
+
+async function fetchDataAndUseData() {
+    await getAllData(currentPage);
+    console.log(dataList); // Now dataList will be populated
+    // You can use dataList here or pass it to another function
+}
+fetchDataAndUseData();
+
+// display posts page 
+function displayItems(dataList, currentPage) {
```

Comment:
seperated class base implementation.

Reply:
Done, made a separate BaseAPI class which is extended by APIService class